### PR TITLE
GET api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by muxappo**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -175,4 +175,72 @@ describe("GET requests", () => {
...
+
+    it("responds with an array of comment objects", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body: { comments } }) => {
+          expect(Array.isArray(comments)).toBe(true);
+          expect(comments.length).toBeGreaterThan(0);
+
+          for (let i = 0; i < comments.length; i++) {
```
Comment (Beth-McCormack) [inline __tests__/integration.test.js]: I'm not sure this test is needed. It is not proving that the response contains any comments, which the test description is suggesting. I think the next test is more suitable

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -175,4 +175,72 @@ describe("GET requests", () => {
...
+            expect(isObject && isNotArray).toBe(true);
+          }
+        });
+    });
+
+    it("returned objects contain the required properties", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .then(({ body: { comments } }) => {
+          return comments.forEach((comment) => {
```
Comment (Beth-McCormack) [inline __tests__/integration.test.js:190]: A length check would be beneficial here, to ensure the response returns the right number of comments

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -175,4 +175,72 @@ describe("GET requests", () => {
...
+
+    it("returned array should be sorted by most recent comments", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .then(({ body: { comments } }) => {
+          expect(comments).toBeSortedBy("created_at", { descending: true });
+        });
+    });
+
+    it("returns a 404 if no comments are found OR if the ID does not exist", () => {
```
Comment (Beth-McCormack) [inline __tests__/integration.test.js]: I think these would be better suited in 2 different tests as the 404 error is being thrown for different reasons

File: __tests__/integration.test.js [hunk 4]
```diff
@@ -175,4 +175,72 @@ describe("GET requests", () => {
...
+      const noArticle = request(app)
+        .get("/api/articles/45/comments")
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Article not found");
+        });
+
+      const noComments = request(app)
```
Comment (Beth-McCormack) [inline __tests__/integration.test.js]: If an article has no comments, is that an error?

File: models/models.js [hunk 5]
```diff
@@ -52,3 +52,33 @@ exports.fetchAllArticles = () => {
...
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          message: "Article not found",
+        });
+      }
+
+      return db
+        .query(
+          `SELECT * FROM comments
```
Comment (Beth-McCormack) [inline models/models.js:69]: Nice query👌

File: controllers/controllers.js [hunk 6]
```diff
@@ -36,3 +37,11 @@ exports.getAllArticles = (req, res, next) => {
     })
     .catch((err) => next(err));
 };
+
+exports.getCommentsByArticle = ({ params: { article_id } }, res, next) => {
+  fetchCommentsByArticle(article_id)
```
Comment (Beth-McCormack) [inline controllers/controllers.js:42]: Controller looks good!

**Review by Beth-McCormack — COMMENTED** [review]
I've left a few comments, please have a look and submit another pr when done🙂

**Review by roisinoreilly — APPROVED** [review]
thanks for refactoring, PR approved :)

**Pull request merged**